cpp/memilio/compartments/feedback_simulation.h

```diff
@@ -0,0 +1,370 @@
+
+namespace mio
+{
+/**
+ * @brief Daily local ICU occupancy aggregated for each age group.
+ */
+template <typename FP = double>
+struct ICUOccupancyLocal {
+    using Type = mio::TimeSeries<FP>;
+    static Type get_default(AgeGroup size)
+    {
+        return Type(size.get());
+    }
+    static std::string name()
+    {
+        return "ICUOccupancyLocal";
+    }
+};
+
+/**
+ * @brief Shape parameter of the gamma distribution.
+ */
+template <typename FP = double>
+struct GammaShapeParameter {
+    using Type = FP;
+    static Type get_default(AgeGroup)
+    {
+        return FP(6);
+    }
+    static std::string name()
+    {
+        return "GammaShapeParameter";
+    }
+};
+
+/**
+ * @brief Scale parameter of the gamma distribution.
+ */
+template <typename FP = double>
+struct GammaScaleParameter {
+    using Type = FP;
+    static Type get_default(AgeGroup)
+    {
+        return FP(0.4);
+    }
+    static std::string name()
+    {
+        return "GammaScaleParameter";
+    }
+};
+
+/**
+ * @brief Number of days in the past considered for the gamma distribution.
+ */
+struct GammaCutOff {
+    using Type = size_t;
+    static Type get_default(AgeGroup)
+    {
+        return 45;
+    }
+    static std::string name()
+    {
+        return "GammaCutOff";
+    }
+};
+
+/**
+ * @brief Maximum allowed contact reduction factors per location.
+ */
+template <typename FP = double>
+struct ContactReductionMax {
+    using Type = std::vector<FP>;
+    static Type get_default(AgeGroup)
+    {
+        return Type(1, FP(1.0));
+    }
+    static std::string name()
+    {
+        return "ContactReductionMax";
+    }
+};
+
+/**
+ * @brief Minimum allowed contact reduction factors per location.
+ */
+template <typename FP = double>
+struct ContactReductionMin {
+    using Type = std::vector<FP>;
+    static Type get_default(AgeGroup)
+    {
+        return Type(1, FP(0.0));
+    }
+    static std::string name()
+    {
+        return "ContactReductionMin";
+    }
+};
+
+/**
+ * @brief Soft-plus curvature parameter for contact adjustment.
+ */
+template <typename FP = double>
+struct SoftPlusCurvatureParameter {
+    using Type = FP;
+    static Type get_default(AgeGroup)
+    {
+        return FP(0.1);
+    }
+    static std::string name()
+    {
+        return "SoftPlusCurvatureParameter";
+    }
+};
+
+/**
+ * @brief Nominal ICU capacity.
+ */
+template <typename FP = double>
+struct NominalICUCapacity {
+    using Type = FP;
+    static Type get_default(AgeGroup)
+    {
+        return FP(9.0);
+    }
+    static std::string name()
+    {
+        return "NominalICUCapacity";
+    }
+};
+
+template <typename FP = double>
+using ParametersFeedback = ParameterSet<ICUOccupancyLocal<FP>, GammaShapeParameter<FP>, GammaScaleParameter<FP>,
+                                        GammaCutOff, ContactReductionMax<FP>, ContactReductionMin<FP>,
+                                        SoftPlusCurvatureParameter<FP>, NominalICUCapacity<FP>>;
+
```

Comment:
I think "FeedbackParameters" or "FeedbackSimulationParameters" would sound nicer, but I don't mind this name too much.

Should we maybe have a `feedback_parameters.h`? Since most other ParameterSets use their own header.